src/states/client.rs

```diff
@@ -337,22 +338,23 @@ impl Bootstrapped for Client {
         }
 
         // Get PeerId of the proxy node
-        let proxy_peer_id = if let Authority::Client { ref proxy_node_name, .. } =
-            routing_msg.src {
-            if *self.proxy_public_id.name() == *proxy_node_name {
-                self.proxy_peer_id
+        let (proxy_peer_id, sending_nodes) =
+            if let Authority::Client { ref proxy_node_name, .. } = routing_msg.src {
+                (if *self.proxy_public_id.name() == *proxy_node_name {
+                     self.proxy_peer_id
+                 } else {
+                     error!("{:?} - Unable to find connection to proxy node in proxy map",
+                            self);
+                     return Err(RoutingError::ProxyConnectionNotFound);
+                 },
+                 iter::once(*self.full_id().public_id()).collect())
```

Comment:
That doesn't need to be part of that large `if` clause, since it's always only going to be that single ID. You could just assign `sending_nodes` in line 356.